main.py

```diff
@@ -1,68 +1,63 @@
-import os
-import time
 import boto3
-import sys
+
 from watchdog.observers import Observer
 from watchdog.events import FileSystemEventHandler
   
-s3_bucket_name = ""
-filename = ""
-  
 class OnMyWatch:
     # Set the directory on watch
     watchDirectory = "e:/Code/indellinent/test/"
   
-    def __init__(self):
+    def __init__(self, handlers):
         self.observer = Observer()
-  
+        self.handlers = handlers
+
     def run(self):
-        event_handler = Handler()
-        self.observer.schedule(event_handler, self.watchDirectory)
+        for handler in self.handlers:
+            self.observer.schedule(handler, self.watchDirectory)
+
         self.observer.start()
         try:
-            while True:
-                time.sleep(5)
-        except:
+            while self.observer.isAlive():
```

Comment:
since this is a "running indefinitely" process add lots of logging to ensure useful diagnostic information?

[loguru](https://pypi.org/project/loguru/) is an easy way to add really simple logging to your app, but it does diverge from using the standard logging library (it seems to be fairly popular though).